config.go

```diff
@@ -1616,6 +1622,35 @@ func ValidateConfig(cfg Config, interceptor signal.Interceptor, fileParser,
 		cfg.DB.Bolt.NoFreelistSync = !cfg.SyncFreelist
 	}
 
+	// Parse any extra sqlite pragma options that may have been provided
+	// to determine if they override any of the defaults that we will
+	// otherwise add.
+	sqlitePragmaOpts := make(
+		map[string]bool, len(cfg.DB.Sqlite.PragmaOptions),
+	)
+	for _, option := range cfg.DB.Sqlite.PragmaOptions {
+		pair := strings.Split(option, "=")
+		if len(pair) != 2 { //nolint:gomnd
+			return nil, fmt.Errorf("invalid sqlite pragma "+
+				"option: %v. Expected the format: name=value",
+				option)
+		}
+
+		sqlitePragmaOpts[pair[0]] = true
+	}
+
+	if !sqlitePragmaOpts["synchronous"] {
+		cfg.DB.Sqlite.PragmaOptions = append(
+			cfg.DB.Sqlite.PragmaOptions, "synchronous=normal",
+		)
+	}
+
+	if !sqlitePragmaOpts["auto_vacuum"] {
+		cfg.DB.Sqlite.PragmaOptions = append(
+			cfg.DB.Sqlite.PragmaOptions, "auto_vacuum=incremental",
+		)
+	}
```

Comment:
Shouldn't these backend-specific defaults live in the sqlite package?

Reply:
originally had it like that but since that package will get tagged and we might want to change these options in future without re-tagging that package, decided to move them here

Reply:
Got it. Is it tagged because it is reused in other projects?

Reply:
No, because it's a submodule and we removed the local `replace` directives. So every change in `kvdb` requires a new tag and then a PR in the main `go.mod` to be pulled in.